src/test/java/org/primeframework/mvc/test/RequestBuilder.java

```diff
@@ -464,6 +466,17 @@ public RequestBuilder withCookie(String name, String value) {
     return this;
   }
 
+  /**
+   * Encrypt the provided value and add a cookie with the encrypted value to the request
+   *
+   * @param name  The name of the cookie.
+   * @param value The unencrypted value of the cookie.
+   * @return This.
+   */
+  public RequestBuilder withEncryptedCookie(String name, String value) throws Exception {
+    return withCookie(name, CookieTools.toCookie(value.getBytes(StandardCharsets.UTF_8), false, true, injector.getInstance(Encryptor.class)));
+  }
```

Comment:
1. While we are in here, should we support compression as well? 
2. `withCookie` supports a `null` value for both the `name` and `value` parameter. This new method does not support `null` for the `value` parameter. Should we keep this consistent, and allow `value` to be `null`? In general, we would expect the callers of this method to have a non-null value, however sometimes when we use testes with data sources or something like that, it can be very convenient to support passing in a `null` value.

Reply:
1. Probably. I didn't add a separate function for that because cookie compression is rarely used (search criteria and `oauth1` state are the only places I see). You'll have to use the version with all the parameters to compress a cookie, but encrypted cookies are fairly common.
2. Modified to support `null` value